line-length does not work with Unicode characters

I have the following UTF-8-encoded document:

```yaml
---
# This is a test to check if “line-length” works nice with unicode characters
check: 1
foo: 'This is another quick test to check behavior with “unicode characters”'
```

Note that the comment in line two as well as the string in line four both contain a Unicode left double quotation mark (U+201C) and a Unicode right double quotation mark (U+201D).  Both lines are 77 characters long, yet yamllint reports “line too long” errors:

```console
% yamllint test.yml
test.yml
  2:81      error    line too long (81 > 80 characters)  (line-length)
  4:81      error    line too long (81 > 80 characters)  (line-length)
% yamllint --version
yamllint 1.13.0
```

It seems as if yamllint counts bytes, not characters, as each Unicode quotation character takes up three bytes when UTF-8 encoded.

_Note: this occurs when running yamllint in a Python 2 environment.  I have not tested it with Python 3, which for my specific use case is not an option yet._

Well seen!

I just tested with both Python 2 and 3: the problem only occurs with Python 2:
```
$ PYTHONPATH=. python3 yamllint/__main__.py /tmp/test.yaml
# nothing
$ PYTHONPATH=. python2 yamllint/__main__.py /tmp/test.yaml
/tmp/test.yaml
  2:81      error    line too long (81 > 80 characters)  (line-length)
  4:81      error    line too long (81 > 80 characters)  (line-length)
```

I'll try to write tests that show which Python versions have the problem.

--- a/tests/rules/test_line_length.py
+++ b/tests/rules/test_line_length.py
@@ -14,6 +14,13 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import sys
+try:
+    assert sys.version_info >= (2, 7)
+    import unittest
+except AssertionError:
+    import unittest2 as unittest
+
 from tests.common import RuleTestCase
 
 
@@ -155,3 +162,16 @@ class LineLengthTestCase(RuleTestCase):
                    'content: |\n'
                    '  {% this line is' + 99 * ' really' + ' long %}\n',
                    conf, problem=(3, 81))
+
+    @unittest.skipIf(sys.version_info < (3, 0), 'Python 2 not supported')
+    def test_unicode(self):
+        conf = 'line-length: {max: 53}'
+        self.check('---\n'
+                   '# This is a test to check if “line-length” works nice\n'
+                   'with: “unicode characters” that span accross bytes! ↺\n',
+                   conf)
+        conf = 'line-length: {max: 52}'
+        self.check('---\n'
+                   '# This is a test to check if “line-length” works nice\n'
+                   'with: “unicode characters” that span accross bytes! ↺\n',
+                   conf, problem1=(2, 53), problem2=(3, 53))

--- a/yamllint/rules/line_length.py
+++ b/yamllint/rules/line_length.py
@@ -17,6 +17,10 @@
 """
 Use this rule to set a limit to lines length.
 
+Note: with Python 2, the ``line-length`` rule may not work properly with
+unicode characters because of the way strings are represented in bytes. We
+recommend running yamllint with Python 3.
+
 .. rubric:: Options
 
 * ``max`` defines the maximal (inclusive) length of lines.

